plugins/roleplay.py

```diff
@@ -0,0 +1,317 @@
+            for t_role in msg.guild.roles:
+                if t_role.id in self.plugin_config[gid]["race_roles"]:
+                    t_s = f"{t_role.name}\n"
+                    if len(t_r)+len(t_s) > 1997:
+                        await respond(msg, t_r+"```")
+                        t_r = t_s
+                    else:
+                        t_r += t_s
+            await respond(msg, t_r+"```")
+        else:
+            if args[1].lower() == "add":
+                if msg.role_mentions:
+                    for role in msg.role_mentions:
+                        if role.id not in self.plugin_config[gid]["race_roles"]:
+                            self.plugin_config[gid]["race_roles"].append(role.id)
+                    await respond(msg, "**AFFIRMATIVE. Roles added to race list.**")
+                else:
+                    raise CommandSyntaxError("No roles mentioned.")
+            elif args[1].lower() == "remove":
+                if msg.role_mentions:
+                    for role in msg.role_mentions:
+                        if role.id not in self.plugin_config[gid]["race_roles"]:
+                            self.plugin_config[gid]["race_roles"].remove(role.id)
+                    await respond(msg, "**AFFIRMATIVE. Roles removed from race list.**")
+                else:
+                    raise CommandSyntaxError("No roles mentioned.")
+            else:
+                raise CommandSyntaxError(f"Unsupported mode {args[1].lower()}.")
+
+    @Command("bio",
+             doc="Adds, edits, prints, dumps or deletes character bios.\n"
+                 "Each character name must be unique.\n"
+                 "Fields: race/gender/height/age: limit 64 characters. theme/link: must be viable http(s) url. "
+                 "appearance/equipment/skills/personality/backstory/interests: limit 1024 characters.\n"
+                 "Setting 'race' to the same name as a registered character role will fetch the colour.\n"
+                 "Be aware that the total length of the bio must not exceed 6000 characters.",
+             syntax="\nediting/creating : (name) set (field) [value]\n"
+                    "printing : (name)\n"
+                    "dumping : (name) dump\n"
+                    "deleting: (name) delete",
+             category="role_play",
+             run_anywhere=True)
+    async def _bio(self, msg):
+        """
+        multipurpose command.
+
+        !bio (name) - print out bio
+        !bio (name) (set) (field) [value] - set a bio value
+        !bio (name) (delete) - delete the bio
+
+        :param msg:
+        :return:
+        """
+        gid = str(msg.guild.id)
+        self._initialize(gid)
+        try:
+            args = shlex.split(msg.content)
+        except ValueError as e:
+            self.logger.warning("Unable to split {data.content}. {e}")
+            raise CommandSyntaxError(e)
+
+        if len(args) < 2:
+            raise CommandSyntaxError("At least one argument required")
```

Comment:
Punctuation

Reply:
Your error message had ane extra dot after that.

Reply:
`f"**WARNING: {err} ANALYSIS: Proper usage: {deco}{command} {fn.syntax}.**"`
I don't think so

Reply:
Now, look at the one without syntax. It just outputs `**WARNING: {err}**`.

Reply:
The message without syntax is incorrect, then. Mostly irrelevant, since all commands that can have syntax errors should have syntax attached to them.